wp-content/themes/genesis-starter/functions.php

```diff
@@ -37,11 +37,20 @@
 
 //* Setup theme
 \add_action( 'after_setup_theme', function () {
-	$scripts    = new \Genesis_Starter\Scripts();
-	$styles     = new \Genesis_Starter\Styles();
-	$shortcodes = new \Genesis_Starter\Shortcodes();
-
-	$scripts->ready();
-	$styles->ready();
-	$shortcodes->ready();
+	$components = array(
+		'scripts'    => new \Genesis_Starter\Scripts(),
+		'shortcodes' => new \Genesis_Starter\Shortcodes(),
+		'styles'     => new \Genesis_Starter\Styles(),
+	);
+
+	// Excluded components
+	$exclude = \apply_filters( 'gs_exclude_components', array() );
```

Comment:
Why not filter the `$components` array instead? That would let you add as well as remove components.

Also, avoid abbreviations, call the hook `genesis_starter_components`.